core/src/main/java/org/infinispan/interceptors/impl/CacheMgmtInterceptor.java

```diff
@@ -695,7 +777,7 @@ public int getNumberOfEntriesInMemory() {
          displayName = "Memory used by data in the cache"
    )
    public long getDataMemoryUsed() {
-      if (cacheConfiguration.memory().isEvictionEnabled() && cacheConfiguration.memory().evictionType() == EvictionType.MEMORY) {
+      if (cacheConfiguration.memory().maxSizeBytes() > 0) {
          return dataContainer.evictionSize();
       }
       return 0;
```

Comment:
Based on `org.infinispan.stats.Stats#getDataMemoryUsed` javadoc, this should return `-1`.

Reply:
Changed